datadog/resource_datadog_dashboard.go

```diff
@@ -305,14 +305,14 @@ func buildTerraformTemplateVariables(datadogTemplateVariables *[]datadogV1.Dashb
 	terraformTemplateVariables := make([]map[string]string, len(*datadogTemplateVariables))
 	for i, templateVariable := range *datadogTemplateVariables {
 		terraformTemplateVariable := map[string]string{}
-		if v, ok := templateVariable.GetNameOk(); ok {
-			terraformTemplateVariable["name"] = *v
+		if v := templateVariable.GetName(); len(v) > 0 {
```

Comment:
Ah sorry, my suggestion only works on prefix, because it's marked nullable. Other fields don't have the same property.

Reply:
No worries :) I reverted the other two as they were before then, given that it's unlikely they'll map to a nil pointer anyway.